contracts/test/EthscriptionsBurn.t.sol

```diff
@@ -0,0 +1,195 @@
+        vm.prank(alice);
+        vm.expectRevert();
+        ethscriptions.transferFrom(address(0), bob, tokenId);
+    }
+    
+    function testBurnUpdatesBalances() public {
+        uint256 tokenId = uint256(testTxHash);
+        
+        // Check initial balance
+        assertEq(ethscriptions.balanceOf(alice), 1);
+        
+        // Burn the token
+        vm.prank(alice);
+        ethscriptions.transferFrom(alice, address(0), tokenId);
+        
+        // Check balance after burn
+        assertEq(ethscriptions.balanceOf(alice), 0);
+        // Note: We can't check balanceOf(address(0)) as OpenZeppelin prevents that
+        // But the token should be burned (not owned by anyone)
+    }
+    
+    function testOnlyOwnerCanBurn() public {
+        uint256 tokenId = uint256(testTxHash);
+        
+        // Bob tries to burn alice's token (should fail)
+        vm.prank(bob);
+        vm.expectRevert();
+        ethscriptions.transferFrom(alice, address(0), tokenId);
+        
+        // Token should still be owned by alice
+        assertEq(ethscriptions.ownerOf(tokenId), alice);
+    }
+    
+    function testApprovedCanBurn() public {
+        uint256 tokenId = uint256(testTxHash);
+        
+        // Alice approves bob
+        vm.prank(alice);
+        ethscriptions.approve(bob, tokenId);
+        
+        // Bob burns the token with approval
+        vm.prank(bob);
+        ethscriptions.transferFrom(alice, address(0), tokenId);
+        
+        // Verify the token is burned
+        vm.expectRevert();
+        ethscriptions.ownerOf(tokenId);
+    }
+    
+    function testBurnCallsTokenManagerHandleTransfer() public {
+        // Create a simple non-token ethscription first to test basic burn
+        bytes32 simpleTxHash = keccak256("simple_tx");
+        Ethscriptions.CreateEthscriptionParams memory params = Ethscriptions.CreateEthscriptionParams({
+            transactionHash: simpleTxHash,
+            initialOwner: alice,
+            contentUri: bytes("data:text/plain,Simple text"),
+            mimetype: "text/plain",
+            mediaType: "text",
+            mimeSubtype: "plain",
+            esip6: false,
+            isCompressed: false,
+            tokenParams: Ethscriptions.TokenParams({
+                op: "",
+                protocol: "",
+                tick: "",
+                max: 0,
+                lim: 0,
+                amt: 0
+            })
+        });
+        
+        vm.prank(alice);
+        ethscriptions.createEthscription(params);
+        
+        // Burn the ethscription
+        vm.prank(alice);
+        ethscriptions.transferFrom(alice, address(0), uint256(simpleTxHash));
+        
+        // Verify it's burned
+        vm.expectRevert();
+        ethscriptions.ownerOf(uint256(simpleTxHash));
+        
+        // The burn should have called TokenManager.handleTokenTransfer with to=address(0)
+        // This ensures TokenManager is notified of burns
```

Comment:
how it's ensured that TokenManager is notified? There's nothing token-related in that test?